conans/test/functional/command/create_test.py

```diff
@@ -516,3 +516,60 @@ def build(self):
         self.assertIn("Build error", client.out)
         package_folder = client.cache.package_layout(pref.ref).package(pref)
         self.assertFalse(os.path.exists(package_folder))
+
+    def create_with_name_and_version_test(self):
+        client = TestClient()
+        conanfile = textwrap.dedent("""
+                from conans import ConanFile
+
+                class MyPkg(ConanFile):
+                    pass
+                """)
+        client.save({"conanfile.py": conanfile})
+        client.run('create . lib/1.0@')
+        self.assertIn("lib/1.0: Created package revision", client.out)
+
+    def create_with_only_user_channel_test(self):
+        """This should be the recommended way and only from Conan 2.0"""
+        client = TestClient()
+        conanfile = textwrap.dedent("""
```

Comment:
conanfile = TestConanFile("lib", "0.1")

``client.save()`` already transforms objects to strings internally

NOT necessary to change it here, just a reminder that we should unify this with GenConanfile and have something useful for testing.

Reply:
https://github.com/conan-io/conan/issues/5525